framework/tests/nontopdefault/NTDTest.java

```diff
@@ -0,0 +1,45 @@
+import tests.nontopdefault.qual.*;
+
+/* Hierarchy:
+ *    NTDTop (default for local variables, implicit upper bound, and receiver)
+ *    /     \
+ * NTDSide  NTDMiddle (default in hierarchy, default for exceptions and resource variables)
+ *   \       /
+ *   NTDBottom (default for implicit and explicit lower bounds, implicit for null literal and Void.class)
+ */
+
+// Because classes and interfaces are by default NTDMiddle, a single override is defined in the
```

Comment:
Everything is by default NTDMiddle, while local variables are NTDTop so `Demand<InnerDefaultReceiver> constructorReference` is actually trying to declare '@NTDTop Demand<@NTDMiddle InnerDefaultReceiver> constructorReference`. The override allows the reference to be declared.